src/android/BLECentralPlugin.java

```diff
@@ -833,15 +847,22 @@ private void autoConnect(CallbackContext callbackContext, String macAddress) {
         addBondStateListener();
 
         if (COMPILE_SDK_VERSION >= 29 && Build.VERSION.SDK_INT >= 29) {
-            LOG.d("testing bonded BLE 1 state in auto", String.valueOf(bondedState));
-            if (bondedState > 0) {
-                LOG.d("testing bonded BLE 1 state in auto here here", "hello");
+            LOG.d(TAG, "testing bonded BLE 1 state in auto" + peripheral.getDevice().getBondState());
+            if (peripheral.getDevice().getBondState() == BluetoothDevice.BOND_BONDED) {
+                LOG.d(TAG, "testing bonded BLE 1 state in auto here here hello");
                 peripheral.connect(callbackContext, cordova.getActivity(), true);
             } else {
                 BluetoothDevice device = bluetoothAdapter.getRemoteDevice(macAddress);
-                LOG.d("testing BLE 3 four YO YO YO", String.valueOf(device));
+                LOG.d(TAG, "testing BLE 3 four YO YO YO" + peripheral.getDevice());
+                setPairingCallback((btDevice, bondedState) -> {
```

Comment:
Creating and setting a callback function to call peripheralDevice.connect() as soon as state changed to BOND_BONDED